src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,8 +3,8 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(TargetFrameworkVersion)' != 'v3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(TargetFrameworkVersion)' != 'v2.1')">
+  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
```

Comment:
This targets file is used in all projects (classic and SDK) so I'm a little nervous about _TargetFrameworkVersionWithoutV being referenced here as it is defined by SDK targets.

(Aside: I'm NOT concerned about the _ ("private") as others are using it and I've basically said elsewhere that it can be considered public at this point. Hoping to get rid of it being heavily used with https://github.com/microsoft/msbuild/issues/3212)

I *think* msbuild short-circuiting would be good enough as we only check this variable when TFI is NETCoreApp or NETStandard, which are only supported for SDK projects, but there were some weird cases of other IDEs design-time evaluating in an "ignore conditions mode" that caused unexpected expressions to be evaluated and complain about numeric comparison to non-number. 

I'll defer to @rainersigwald on this.

Reply:
My preference is we only depend on publicly supported values. If that means we have to add another `AND` for every version of .NET Core app for the foreseeable future than so be it. The total amount of code that will take is less than the comment section of this PR.

Reply:
The problems with short-circuiting were only when an outer scope was conditioned but the inner scope isn't; that's not the case here--the short-circuiting is within a single expression.

I manually applied this patch to my local VS and was able to load a project just fine using OmniSharp, which was the major source of problems there. I think it's good.

Reply:
@nguerrera Can you elaborate a little on the problem with `_TargetFrameworkVersionWithoutV`? This is in the CSharp targets so I assumed the SDK would always be available. Is that a bad assumption?

Reply:
By SDK here (overloaded term), I mean Microsoft.NET.Sdk and its targets. In a classic csproj (no Sdk="Microsoft.NET.Sdk"), this file would be imported but the Microsoft.NET.Sdk wouldn't. And Microsoft.NET.Sdk defines _TargetFrameworkVersionWithoutV. But we are only checking _TargetFrameworkWithoutV when target framework .NETCoreApp or .NETStandard, which would require Microsoft.NET.Sdk.

Reply:
This change broke our repo with VS 2019 16.4.0 preview 2.0. Our repo uses classic/non-sdk csproj to target netstandard. I worked around by defining `_TargetFrameworkVersionWithoutV` in a props file.

Reply:
@timmydo Is there a reason why you're targeting .NET Standard using non-SDK projects?

Reply:
Long story, but we migrated from corext to stock msbuild and I think this was the recommended approach. Maybe we aren't completely there yet, but when you have thousands of csproj files, it's not like an on-off switch you can flip to move to the latest thing. There are probably a lot of others in the same boat.

Reply:
Interesting that it would be a recommended approach. IIRC we don't support projects in that state, though I could be wrong.